flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -482,23 +482,25 @@ public void requestPartitions() throws IOException, InterruptedException {
 		}
 
 		synchronized (requestLock) {
-			if (isReleased) {
-				throw new IllegalStateException("Already released.");
-			}
+			if (!requestedPartitionsFlag) {
```

Comment:
we should follow the same "having the shorter/simpler branch first" approach as before (also see https://github.com/apache/flink/pull/6974#discussion_r229666072 )